worker.py

```diff
@@ -140,34 +148,40 @@
 scene.add(lamp_fill)
 
 
-# TODO: this is a hack. This conversion should be done automatically and internally.
+# TODO: this is a hack; conversion should be done automatically.
 def translate_quat(pb_quat):
   """ Convert pyBullet XYZW quaternions into Blender WXYZ quaternions."""
   x, y, z, w = pb_quat
   return [w, x, y, z]
 
 
 # --- Dump the simulation data in the renderer
-room = scene.add_from_file(str(floor.vis_filename))
-room.position = floor.position
-room.quaternion = translate_quat(floor.rotation)
-
+floor_mesh = THREE.Mesh.from_file(str(floor.vis_filename))
+floor_mesh.position = floor.position
+floor_mesh.quaternion = translate_quat(floor.rotation)
+scene.add(floor_mesh)
 
 for obj in objects:
-  o = scene.add_from_file(str(obj.vis_filename), name=obj.uid)
-  o.position = obj.position
-  o.quaternion = translate_quat(obj.rotation)
-
-  if FLAGS.randomize_color:
-    logging.warning("TODO: color randomization")
-    pass
-
+  # --- Load the mesh into the scene
+  mesh = THREE.Mesh.from_file(str(obj.vis_filename), name=obj.uid)
+  mesh.position = obj.position # TODO: why set? see keyframing below
```

Comment:
Both in our code and in Blender there is a distinction between the "default" position of an object and the keyframed position. It is a bit strange, but seems to be pretty standard. 
For rendering the animation this is irrelevant though.